packages/cactus-plugin-satp-hermes/src/test/typescript/unit/recovery/logging.test.ts

```diff
@@ -0,0 +1,376 @@
+import "jest-extended";
+import {
+  CrashRecoveryManager,
+  CrashStatus,
+} from "../../../../main/typescript/core/recovery/crash-manager";
+import { LogLevelDesc, Secp256k1Keys } from "@hyperledger/cactus-common";
+import { ICrashRecoveryManagerOptions } from "../../../../main/typescript/core/recovery/crash-manager";
+import knex from "knex";
+import {
+  LocalLog,
+  SupportedChain,
+} from "../../../../main/typescript/core/types";
+import {
+  Asset,
+  CredentialProfile,
+  LockType,
+  SignatureAlgorithm,
+} from "../../../../main/typescript/generated/proto/cacti/satp/v02/common/message_pb";
+import { v4 as uuidv4 } from "uuid";
+import { SessionData } from "../../../../main/typescript/generated/proto/cacti/satp/v02/common/session_pb";
+import { SATP_VERSION } from "../../../../main/typescript/core/constants";
+import { SATPSession } from "../../../../main/typescript/core/satp-session";
+import { knexClientConnection } from "../../knex.config";
+import { getSatpLogKey } from "../../../../main/typescript/gateway-utils";
+import { TokenType } from "../../../../main/typescript/core/stage-services/satp-bridge/types/asset";
+import { RecoverUpdateMessage } from "../../../../main/typescript/generated/proto/cacti/satp/v02/crash_recovery_pb";
+
+const logLevel: LogLevelDesc = "DEBUG";
+
+let mockSession: SATPSession;
+const keyPairs = Secp256k1Keys.generateKeyPairsBuffer();
+const sessionId = uuidv4();
+
+const createMockSession = (maxTimeout: string, maxRetries: string) => {
+  const mockSession = new SATPSession({
+    contextID: "MOCK_CONTEXT_ID",
+    server: false,
+    client: true,
+  });
+
+  const sessionData = mockSession.hasClientSessionData()
```

Comment:
We need to consider this carefully. If sessionData remains as it is, we must handle it with care and clearly differentiate between the client and server sides of the gateway. I designed the sessionData this way to ensure that a gateway can act as both a client and server to itself.

Reply:
@RafaelAPB